### DISCUSSION TITLE
Search Query on multiple words

### DISCUSSION THREAD

**Discussion by lopcheung123** [microsoft-search/pnp-modern-search#1737] [General]
Hello,

I am having issues getting the search to work the way I need it too in my tenant. When I try to search multiple words "Payment Card", it looks for the exact phrase and not each word, "Payment" Or "Card". I tried to look for help thru the old issues but was not able to find any help w/ how to make it work this way.

Would like some guidance if there is a way to make it search short phrases/sentences, searching each individual word as well.

Thanks

**Comment by wobba** [comment]
`payment card` <- `payment AND card` in any order
`"payment card"` <- exact phrase

See https://docs.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference#using-phrases-in-the-free-text-kql-query

**Reply by lopcheung123** [reply]
Thanks for the link.  The default is currently {searchTerms}.  So to  get the above working as you described, I would need to edit the kql under "Query Template" box?  
image below
![searchsnippet](https://user-images.githubusercontent.com/97464703/150850016-66fd792b-3805-45c6-be19-333cc93091ec.PNG)

I'm new to configuring search =] Appreciate your help

**Reply by wobba** [reply]
Not really.. just don't add quotes when you type in the search box. It will be an implicit AND between terms entered. If you add quotes it's a phrase.

**Reply by StephenHoughton4** [reply]
Hi Wobba, Is it possible to make the Search term use OR? for example Typing "James Dublin" (without quotes) Would find people called james in the Dublin Office (as well as anyone called James Dublin)? 

I had the QueryTemplate as:
AccountName:({searchTerms}*) OR Office:({searchTerms}*)

in debug the queryModification shows:
"queryModification": "AccountName:(James dublin*) OR Office:(James dublin*) 

I would want AccountName to find James* and Office to find Dublin* (and if I expanded LastName to find James* or Dublin*)

I tried initially for the QueryTemplate as ({searchTerms}*)
but this brought back people that are not James or Dublin related..

Thanks
Stephen

**Reply by StephenHoughton4** [reply]
I saw an answer on another thread.. So need to go back to the drawing board anyway..
https://github.com/microsoft-search/pnp-modern-search/discussions/2909

**Comment by lopcheung123** [comment]
I think I know the problem but unsure of what settings I need to get it to work.

When I search using url fragment, it includes the url-encode.  

![paymenttermsearch](https://user-images.githubusercontent.com/97464703/151018352-5db93744-cf98-4c02-9c4e-9e0eac666a1b.PNG)

When I use it as default search, in edit settings, it doesn't include the space (%20) and searches correctly.

![searchterms2](https://user-images.githubusercontent.com/97464703/151018171-0012362c-af39-4899-954b-57c9ae37762e.PNG)


I'll try searching how to get this to work but if you know how I can do this, please let me know.  
Thank you!

**Reply by lopcheung123** [reply]
I am using version 4.1.0.0

release date April 2001

**Reply by lopcheung123** [reply]
I have updated the version to 4.5.3 (latest version) and although the search does not include the encode from the screen shot

![image](https://user-images.githubusercontent.com/97464703/151066688-f89938d1-db5b-46e1-8f0a-4758400d2012.png)


It still returns no results.

If I put in payment terms for the default under edit properties, it returns the results correctly.

**Reply by wobba** [reply]
Well... it means that the query `payment terms IsDocument:1 -fileextension:aspx` does not match anything. Adding the query in "default query" or in the search box should show the same results. If you use the debug display template you can look in the `queryModification` what the executed query was.

![image](https://user-images.githubusercontent.com/51104/151246846-f0548487-c0a3-4716-ac41-56bdf071a49a.png)

**Reply by lopcheung123** [reply]
Thanks for your help.
